llms/mlx_lm/utils.py

```diff
@@ -174,22 +217,29 @@ def generate(
                 print(s[skip:], end="", flush=True)
                 skip = len(s)
 
+        repetition_context.append(
+            token.item()
+        )  # Update repetition context after each token
+        repetition_context = repetition_context[
+            -repetition_context_size:
+        ]  # Maintain the specified context size
+
     token_count = len(tokens)
-    token_string = tokenizer.decode(tokens).replace(REPLACEMENT_CHAR, "")
+    tokens = tokenizer.decode(tokens).replace(REPLACEMENT_CHAR, "")
```

Comment:
issue: `token_string` is a better name because it represents the decoded tokens, not longer tokens but rather a string.